### ISSUE TITLE
[BUG] DLSS swapper 1.0.3.1 does not find my MSFS2020

### ISSUE THREAD

**Issue by AlbertSas** [beeradmoore/dlss-swapper#177]
**Describe the bug**
DLSS swapper 1.0.3.1 does not find my MSFS2020 

**To Reproduce**
1. Go to GAMES tab'
2. Refresh'

**Expected behavior**
See that the game or the dlss file have been found

**Screenshots**

**Additional context**

**Comment by stefanbode** [comment]
Install from dvd and now the ms store it works for me. Seems to be a special configuration thing. What are you using? Steam?

**Comment by Schlicki** [comment]
> Install from dvd and now the ms store it works for me. Seems to be a special configuration thing. What are you using? Steam?

Nope, MS Store version. Starfield and Forza for example is recognized.

**Comment by AlbertSas** [comment]
MS Store version for me as well

 

Van: Schlicki ***@***.***> 
Verzonden: maandag 4 december 2023 20:53
Aan: beeradmoore/dlss-swapper ***@***.***>
CC: AlbertSas ***@***.***>; Author ***@***.***>
Onderwerp: Re: [beeradmoore/dlss-swapper] [BUG] DLSS swapper 1.0.3.1 does not find my MSFS2020 (Issue #177)

 

Install from dvd and now the ms store it works for me. Seems to be a special configuration thing. What are you using? Steam?

Nope, MS Store version. Starfield and Forza for example is recognized.

**Comment by beeradmoore** [comment]
Did any big updates come out in the last few months for MSFS2020? A few issues poped up about the same issue.

User in #173 is also having an issue with MSFS2020, but theirs seemed to be a different issue somehow.

Also in #140 the user had the same issue. Does the information posted in there help?

From memory MSFS2020 (plesae correct me if I am wrong) is a special case where the game you download is actually somewhat of a game manager. So that other tool then downloads MSFS2020. Do you have any additional options of where to install?

Do you know the exact drive it is installed to? For any MSStore/Xbox games to work they need to be installed via Xbox Apps custom install directories. Reason for this is apps by default are installed into `[drive]:\Program Files\WindowsApps\` which is a very locked down directory. Using the custom game stuff (mentioned in issue 140 linked above) it will still install there but it does some magic where there are read-write files accessible outside of there.

To find where your games should be if this is the case, in the root of the drive you installed that game to there should be a file `.GamingRoot`. Inside that file it has this
![image](https://github.com/beeradmoore/dlss-swapper/assets/904737/19000b57-550c-4a27-b420-8f03b5ab7949)

Ignoring the red and the first `RGBX` characters it says `XboxGames`. This is because I install my games into `C:\XboxGames\`. For Xbox/MSStore games to work they need to be installed here (or same method to find path on other drives). If the game is not here we can't detect it.

**Comment by bluememon** [comment]
@beeradmoore tomorrow there's gonna be a big sim update, I'm not really sure if it's gonna change any install structure. I'm on the same place most of my games appear correctly on the list except for msfs.

In my case I have the MS Store Version, since the game gets really big I moved it into another hard drive

**Comment by beeradmoore** [comment]
I don't know when I'll get time, but I'd have to see if the game installs differnetly if installed via the MSStore or via Xbox App. 

Another possible thought that requires more testing is if the game is installed via an Xbox app in a custom directory and then moved to a different drive via Settings app, does it then become a standard `WindowsApp` app and if so that would also explain it breaking. 

Had it ever shown for you do you know?

**Comment by beeradmoore** [comment]
I have A Plague Tale: Requiem installed in `C:\XboxGames\A Plague Tale- Requiem - Windows\` and it shows up like so,
![Screenshot 2023-12-05 104118](https://github.com/beeradmoore/dlss-swapper/assets/904737/a1410d4c-eff7-4d24-b1bb-f13136381a04)

Going to system setings and moving it to `D:\` drive. I do have an XboxApp library setup on this drive, and it did indeed move to `D:\XboxGames\A Plague Tale- Requiem - Windows\` and it is still detected by DLSS Swapper.

Moved it back to `C:\`. Removed the XboxApp library feature (removed `.GamingRoot` and `XboxGames` folder) from `D:\` drive and tried to move it from `C` to `D`. It re-created the `.GamingRoot` and `XboxGames`. Maybe there is a correct way to remove it?

**Comment by Schlicki** [comment]
I moved MSFS in the XBox-App from C:XboxGames to J:XboxGames and now it shows up in the swapper.

**Comment by maucinti** [comment]
MSFS could be installed in various locations depending on the way one bought it....
It's a mess. in my case it's on :
G:\WpSystem\S-1-5-21-191972008-2729848689-3515026992-1000\AppData\Local\Packages\Microsoft.FlightSimulator_8wekyb3d8bbwe
while DLSS files are in a hidden,unaccessible folder 
G:\WindowsApps\
We must thank Microsoft for this brainy, useless ,stupid configuration...

**Comment by AlbertSas** [comment]
I’m totally unsure :

 

I tried to locate via regedit , did not work

My desktop shortcut refers to :

Microsoft.Flightsimulator_8wekyb3d8bbwe!App

 

Sorry to not be really helpful

 

Regards

Albert

 

maucinti ***@***.***> 
Verzonden: dinsdag 5 december 2023 18:32
Aan: beeradmoore/dlss-swapper ***@***.***>
CC: AlbertSas ***@***.***>; Author ***@***.***>
Onderwerp: Re: [beeradmoore/dlss-swapper] [BUG] DLSS swapper 1.0.3.1 does not find my MSFS2020 (Issue #177)

 

MSFS could be installed in various locations depending on the way one bought it....
It's a mess. in my case it's on :
G:\WpSystem\S-1-5-21-191972008-2729848689-3515026992-1000\AppData\Local\Packages\Microsoft.FlightSimulator_8wekyb3d8bbwe
while DLSS files are in a hidden,unaccessible folder
G:\WindowsApps
We must thank Microsoft for this brainy, useless ,stupid configuration...

**Comment by legranja** [comment]
I am having the same issue.  can not get MSFS 2020 to appear in DLSS swapper 1.0.2.0 under Game, refresh and filter does not help.

**Comment by beeradmoore** [comment]
Just as a sanity check, can you all check settings and ensure xbox app is toggled on?

**Comment by Schlicki** [comment]
> I am having the same issue. can not get MSFS 2020 to appear in DLSS swapper 1.0.2.0 under Game, refresh and filter does not help.

Have you tried to switch install folder in the Xbox app?

**Comment by beeradmoore** [comment]
For ease of use for people that is also `%LOCALAPPDATA%\Packages\Microsoft.FlightSimulator_8wekyb3d8bbwe\` in explorer. But without taking ownership of that folder and potentially ruining everything (at least in my experience 😅) you can't do anything there.

This is why custom install directories via Xbox App is the only way to do it as it does some magic and has files in a read/write folder as well as read/only normal install. Unsure if MSFS2020 is doing funky stuff and not doing what normal games do and making this be a problem or what.

**Comment by legranja** [comment]
thanks Brad

Luis Eduardo Granja Vernimmen
________________________________

**Comment by nwenkel** [comment]
The point seems to be that without changing the access rights to the windowsapp folder it´s not possible to have access to exchange files.

**Comment by beeradmoore** [comment]
So I hit download in Xbox app and it got did a small-ish download. Launching the game presented this,
![Screenshot 2023-12-10 112358](https://github.com/beeradmoore/dlss-swapper/assets/904737/3f0b2267-c48d-4442-a3aa-e81d56dcb9d1)

I let it install the 180gb and launched the game.
![Screenshot 2023-12-10 140937](https://github.com/beeradmoore/dlss-swapper/assets/904737/f20b59ae-ead6-42ee-b084-f8bec1ddc7af)

It shows up in DLSS Swapper.

Can anyone recommend steps to get the game in an alternate way so I can diagnose the issues. More than happy to buy it and install from the Windows Store, but last time I installed a purchased game from Windows Store it installed into my defined XboxGames folder as if it was any other game from GamePass.

**Comment by beeradmoore** [comment]
Does anyone else having the problem want to try uninstall and re-install and see if it does the same thing again? This should be a good way to test install methods havn't changed since launch and now it will work for everyone on a fresh install.

**Comment by maucinti** [comment]
Got it.
I played around a bit with Xbox app and got it to install in C:\xbox. In reality, it moves only 2GB (don't know exactly what) to the new location, and it is now recognized by the Swapper. 
Anyone has to search its own way to get this , as every installation coa be different according to initial conditions...

**Comment by Schlicki** [comment]
As I already said: you have to change the install location via the Xbox app to get the MSFS visible for the swapper.

**Comment by beeradmoore** [comment]
Closing this issue
